### PR TITLE
docs(icon): refactor popover

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

Rebuilt icon popover to use `dt-popover` and improve accessibility and overall look.

** Update **

Completely refactored icon search logic trying to improve the performance but still have performance issues... I've tried a dozen different ways to improve performance but something is broken. 

**Notes**
- Improving `dt-popover` with dynamic external anchor might help with performance (POC in progress). 
- Noticed performance issues with a lot of `dt-icon` components on screen too, might need to refactor this to load all icons at once on dialtone-vue and just switch to the needed icon on `dt-icon` component.

Screenshot 2023-01-06 at 6 37 45 p m

Screenshot 2023-01-06 at 6 38 21 p m

Screenshot 2023-01-06 at 6 40 15 p m

Screenshot 2023-01-06 at 6 40 31 p m

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [ ] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: docs/.vuepress/baseComponents/IconPopoverContent.vue [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          :key="`size-${option}`"
+          :selected="option === selectedSize"
+          :value="option"
+          v-text="option"
+        />
+      </dt-select-menu>
+    </div>
+    <div class="d-d-flex d-ai-center">
+      <div class="d-d-flex d-fd-column d-fl-grow5">
+        <span class="d-fc-disabled">Name</span>
```
Comment (francisrupert) [inline docs/.vuepress/baseComponents/IconPopoverContent.vue]: Change to `class="d-label d-label--sm"`

File: docs/.vuepress/baseComponents/IconPopoverContent.vue [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        {{ iconName }}
+      </div>
+      <copy-button
+        class="d-ml8"
+        :text="iconName"
+      />
+    </div>
+    <div class="d-d-flex d-fd-column">
+      <span class="d-fc-disabled">Category</span>
```
Comment (francisrupert) [inline docs/.vuepress/baseComponents/IconPopoverContent.vue]: Change to `class="d-label d-label--sm"`

File: docs/.vuepress/baseComponents/IconPopoverContent.vue [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      <span
+        class="d-tt-capitalize"
+        v-text="category"
+      />
+    </div>
+    <div class="d-d-flex d-fd-column">
+      <span class="d-fc-disabled">Keywords</span>
```
Comment (francisrupert) [inline docs/.vuepress/baseComponents/IconPopoverContent.vue]: Change to `class="d-label d-label--sm"`

File: docs/.vuepress/baseComponents/IconPopoverContent.vue [hunk 4]
```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="d-d-flex d-ai-center d-fl-center d-h64 d-mb24">
```
Comment (francisrupert) [inline docs/.vuepress/baseComponents/IconPopoverContent.vue]: Change `d-mb24` to `d-mb16`

File: docs/.vuepress/baseComponents/IconPopoverContent.vue [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      <span class="d-fc-disabled">Keywords</span>
+      {{ keywords.join(', ') || '-' }}
+    </div>
+    <div class="d-d-flex d-ai-flex-end">
+      <div class="d-fl-grow1">
+        <dt-input
+          label="SVG"
+          readonly
+          tabindex="-1"
+          size="sm"
```
Comment (francisrupert) [inline docs/.vuepress/baseComponents/IconPopoverContent.vue:55]: Add `class="d-ff-mono"`

File: docs/.vuepress/baseComponents/IconPopoverContent.vue [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        :text="svgExample"
+      />
+    </div>
+    <div class="d-d-flex d-ai-flex-end">
+      <div class="d-fl-grow1">
+        <dt-input
+          label="Vue"
+          tabindex="-1"
+          readonly
+          size="sm"
```
Comment (francisrupert) [inline docs/.vuepress/baseComponents/IconPopoverContent.vue:72]: Add `class="d-ff-mono"`

**Review by francisrupert — CHANGES_REQUESTED** [review]
This popover rocks. Great job getting it built! A few tiny formatting changes called out on specific lines. And then a few not viewable in code diff:

* Change `class="d-popover__header d-pl24"` to `class="d-popover__header d-pl16"`.
* Change `class="d-popover__content d-p24"` to `class="d-popover__content d-p16"`

I provisionally approve from design perspective, defer to @braddialpad and @josedialpad for everything else.

File: docs/.vuepress/baseComponents/IconPopover.vue [hunk 7]
```diff
@@ -0,0 +1,68 @@
+<template>
+  <dt-popover
+    :open="modelValue"
+    :modal="true"
```
Comment (braddialpad) [inline docs/.vuepress/baseComponents/IconPopover.vue:4]: Hmm something weird happening with the modality on this.. I'm not able to scroll the page which is correct, but I am able to mouseover and click items outside the modal. Is the invisible overlay possibly not working on the docsite?

File: docs/.vuepress/baseComponents/IconPopoverContent.vue [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        />
+      </dt-select-menu>
+    </div>
+    <div class="d-d-flex d-ai-center">
+      <div class="d-d-flex d-fd-column d-fl-grow5">
+        <span class="d-fc-disabled">Name</span>
+        {{ iconName }}
+      </div>
+      <copy-button
+        class="d-ml8"
```
Comment (braddialpad) [inline docs/.vuepress/baseComponents/IconPopoverContent.vue:32]: can set a copy aria-label on these copy buttons to say specifically what they are copying (ex: copy name, copy SVG markup, copy vue markup)

File: docs/.vuepress/views/IconCatalog.vue [hunk 9]
```diff
@@ -88,74 +89,100 @@
...
+      if (Object.keys(filteredCategory).length) {
+        Object.assign(acc, { [category]: Object.freeze(filteredCategory) });
+      }
+      return acc;
+    }, {});
 
-const isCategoryInResults = (category) => {
-  return Object.keys(iconsList.value).includes(category);
+  filteredIconsList.value = Object.freeze(filteredIcons);
 };
```
Comment (braddialpad) [inline docs/.vuepress/views/IconCatalog.vue]: Lets comment this a bit, it's pretty complex.

File: docs/.vuepress/views/IconCatalog.vue [hunk 10]
```diff
@@ -88,74 +89,100 @@
...
+const delayTimer = ref(null);
+const selectedIcon = ref(undefined);
+
+const searchByIconName = () => {
+  clearTimeout(delayTimer.value);
+  delayTimer.value = setTimeout(() => {
+    searching.value = true;
+    resetCategory();
+    filterIconList();
+  }, 300);
```
Comment (braddialpad) [inline docs/.vuepress/views/IconCatalog.vue]: We should make a debounce utility function for this https://www.freecodecamp.org/news/javascript-debounce-example/

**Review by braddialpad — COMMENTED** [review]
Great work!

File: docs/.vuepress/baseComponents/IconPopover.vue (hunk 7 above)
Comment (juliodialpad) [reply docs/.vuepress/baseComponents/IconPopover.vue:4]: Looks like it wasn't working due to the `d-modal--transparent` element didn't had a proper sizing. 

Will add a fix for this making the overlay full screen.

Screenshot 2023-01-16 at 1 44 19 p m

**Review by braddialpad — APPROVED** [review]
Thanks!

**Review by francisrupert — APPROVED** [review]

**Comment by francisrupert** [comment]
Awesome. Merge away!

**Pull request merged**